classes/task/failed_task.php

```diff
@@ -31,6 +31,98 @@ public function get_name() {
         return get_string('taskfailed', 'logstore_xapi');
     }
 
+    private function extract_failed_events($limitnum) {
+        global $DB;
+        $conditions = null;
+        $sort = '';
+        $fields = '*';
+        $limitfrom = 0;
+        $extractedevents = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+        return $extractedevents;
+    }
+
+    private function extract_events($limitnum) {
+        global $DB;
+        $conditions = null;
+        $sort = '';
+        $fields = '*';
+        $limitfrom = 0;
+        $extractedevents = $DB->get_records('logstore_xapi_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+        return $extractedevents;
+    }
+
+    private function insert_failed_events_into_xapi_log($events) {
+        global $DB;
+        $DB->insert_records("logstore_xapi_log", $events);
+    }
+
+    private function delete_failed_events($events) {
+        global $DB;
+        $eventids = $this->get_delete_event_ids($events);
+        $DB->delete_records_list('logstore_xapi_failed_log', 'id', $eventids);
+    }
+
+    private function get_failed_events($events) {
+        $nonloadedevents = array_filter($events, function ($loadedevent) {
+            return $loadedevent['loaded'] === false;
+        });
+        $failedevents = array_map(function ($nonloadedevent) {
+            return $nonloadedevent['event'];
+        }, $nonloadedevents);
+
+        // enrich failed event with errortype and response
```

Comment:
code styling issue at Inline comments.
Inline comments must use the "// " (2 slashes + whitespace) style, laid out neatly so that it fits among the code and lines up with it. The first line of the comment must begin with a capital letter (or a digit, or '...') and the comment must end with a proper punctuation character. Permitted final characters are '.', '?' or '!'.

Reply:
I have fixed this and will commit the change later.